lib/arrays.md

```diff
@@ -5,44 +5,113 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is an Array?
 
-Replace me with your response...
+An array is a list of values, with each slot, or element, holding a specific value.
+
 
 # What are some examples of information that would be Arrays as opposed to some other data type?
 
-Replace me with your response...
+Lists of information that would need to be independently referenced, and consist of differing datatypes, would be in arrays.
+
 
 # What is one way, using Ruby, to retrieve the 6th element in an Array? How about the 8th element? What happens if you try to retrieve the value of the _9999th_ element (or some element that doesn't exist in the array)?
 
-Replace me with your response...
+Code to retrieve 6th element of an array:
+1) array1 = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j']
+2) puts array1[5]
+3) Returns "f"
+
+Code to retrieve 8th element of an array:
+1) array1 = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j']
+2) puts array1[7]
+3) Returns "h"
+
+Code to retrieve 9,999th element of an array:
+1) array1 = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j']
+2) puts array1[9998]
+3) Returns nil
+
 
 # The previous question asks about finding, for example, the 6th element in an Array. Is it possible to find the **-6th** (Notice the negative symbol!) element in an Array? What does that even mean?
 
-Replace me with your response...
+It is possible to find the -6th element of an array. When counting using a negative number, the program will count from the last element of the array backwards, with the last element having a position of -1 (instead of a 0 like when you count from the beginning of an array). Using the array from the previous question, the array would resolve like so:
+
+Code to retrieve -6th element of an array:
+1) array1 = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j']
+2) puts array1[-6]
+3) Returns "e"
+
+
 
 # How would you perform an operation on every element inside an Array?
 
-Replace me with your response...
+In the case of an array where all the elements are numbers, you can:
+1) define the array, 
+2) use the ".each" method to specify that you what follows the method is to be performed for each array element,
+3) define a variable in verticle bars that will be used as a variable for each iteration of the method, and
+4) type out the operation to be performed for each array element.
+
+For my example, I want each array element to be multiplied by the number 2, and the results displayed:
+array2 = [1, 2, 3, 4, 5, 6, 7, 8]
+array2.each {|z| puts z * 2}
+
```

Comment:
Yep! Nice work! Just to clarify, you can use `each` on Arrays with all kinds of information inside them, but the block of code you run in your `each` loop (here I see you're multiplying the element by 2) will have to be appropriate for the data types in the Array.